### PR TITLE
If there's a collision, change name of URL

### PR DESCRIPTION
**Pull request by dragon-dxw**
Possibly worth a discussion about the shape of collision URLs -- should we just use UUIDs?

### CODE REVIEW

File: ds-caselaw-ingester/lambda_function.py [hunk 1]
```diff
@@ -389,44 +384,45 @@ def handler(event, context):
...
     copy_file(
         tar,
         f"{consignment_reference}/{docx_filename}",
-        f'{uri.replace("/", "_")}.docx',
-        uri,
+        f'{target_uri.replace("/", "_")}.docx',
+        target_uri,
         s3_client,
     )
+    # TODO: the editor move functionality should change the name of the docx file to match?
```
Comment (anthonyhashemi) [inline ds-caselaw-ingester/lambda_function.py:416]: oh that's a good question. Probably makes sense to and also to do so when we move the document in in the merge flow in editor ui?

File: ds-caselaw-ingester/lambda_function.py [hunk 2]
```diff
@@ -439,15 +435,17 @@ def handler(event, context):
...
 
     # Copy original tarfile
-    store_file(open(filename, mode="rb"), uri, os.path.basename(filename), s3_client)
+    store_file(
+        open(filename, mode="rb"), target_uri, os.path.basename(filename), s3_client
+    )
 
-    if api_client.get_published(uri):
-        update_published_documents(uri, s3_client)
+    if api_client.get_published(target_uri):
```
Comment (anthonyhashemi) [inline ds-caselaw-ingester/lambda_function.py:447]: Can we get rid of this block now that we should never be updating documents anymore, just like you removed the marklogic update calls above?

File: ds-caselaw-ingester/lambda_function.py [hunk 3]
```diff
@@ -315,6 +298,18 @@ def insert_document_xml(uri, xml) -> bool:
         return False
 
 
+def select_uri(source_uri):
+    """If the judgment exists, suggest a URL that is not going to be collided with,
+    even if there are multiple requests a second or the ingester is restarted."""
+    global collision_counter
```
Comment (anthonyhashemi) [inline ds-caselaw-ingester/lambda_function.py:304]: If there are multiple invocations of the lambda a second, are they run in the same runtime environment? I would imagine theyd be in different runtimes and so the global variable wouldnt be shared. I'd suggest that we could add a hash or a unique identifier from the lambda run /event message or something along those lines? TDR ref even?

Comment (dragon-dxw) [reply ds-caselaw-ingester/lambda_function.py:304]: ... that's a very valid point

File: ds-caselaw-ingester/lambda_function.py (hunk 2 above)
Comment (dragon-dxw) [reply ds-caselaw-ingester/lambda_function.py]: It took me a while to understand that we don't ever need to worry about the target URI being published, so this is  indeed all rubbish. 🔥

File: ds-caselaw-ingester/lambda_function.py (hunk 1 above)
Comment (dragon-dxw) [reply ds-caselaw-ingester/lambda_function.py:399]: This should probably block the merging of this.

Comment (jacksonj04) [reply ds-caselaw-ingester/lambda_function.py:399]: We should probably note that this is yet another good reason to distinguish between:

- Our own internal identifier for a document, which is really just a UUID, and
- Our 'TNA' identifier for a document, which is an arbitrary string we use in URIs, and
- Other identifiers people may use such as NCNs or Bailii numbers

Comment (dragon-dxw) [reply ds-caselaw-ingester/lambda_function.py:399]: We do actually do this; I had to double check.

**Review by anthonyhashemi — APPROVED** [review]

**Pull request merged**